src/Site_Command.php

```diff
@@ -637,93 +637,136 @@ function( $blog ) {
 	 *
 	 * ## OPTIONS
 	 *
-	 * <id>...
-	 * : One or more IDs of sites to archive.
+	 * [<id>...]
+	 * : One or more IDs of sites to archive. If not provided, you must set the --slug parameter.
+	 *
+	 * [--slug=<slug>]
+	 * : Path of the blog to archive. Subdomain on subdomain installs, directory on subdirectory installs.
 	 *
 	 * ## EXAMPLES
 	 *
 	 *     $ wp site archive 123
 	 *     Success: Site 123 archived.
 	 */
-	public function archive( $args ) {
-		$this->update_site_status( $args, 'archived', 1 );
+	public function archive( $args, $assoc_args ) {
+		if ( isset( $assoc_args['slug'] ) ) {
+			$blog = $this->get_site_by_slug( $assoc_args['slug'] );
+			$this->update_site_status( [ $blog->blog_id ], 'archived', 1 );
+		} else {
+			$this->update_site_status( $args, 'archived', 1 );
+		}
```

Comment:
For all of these, we also need to provide some error mechanism when neither `<id>` nor `--slug=` are provided. Check out `wp plugin activate` and its sibling commands for an example of how this can be implemented:

https://github.com/wp-cli/extension-command/blob/6a68d247969b8968b007a4d664e0de14bf69e775/src/Plugin_Command.php#L317-L320
https://github.com/wp-cli/extension-command/blob/6a68d247969b8968b007a4d664e0de14bf69e775/src/WP_CLI/ParsePluginNameInput.php#L9-L47

We'll want to add tests for this aspect too.

Your `get_site_by_slug()` method could be updated to `$this->get_sites( $args, $assoc_args )` and return a `$blogs` variable that's passed directly into `$this->update_site_status()`

Reply:
Update `get_site_by_slug()` method to `get_sites_ids()`

The check to ensure that the site IDs or slug are provided is now done in the `check_site_ids_and_slug()` method.